### PR TITLE
[19.07]luci-app-attendedsysupgrade: sync with master branch

### PR DESCRIPTION
**Pull request by aparcar**
Upgrade the app to stay compatible with the running upgrade server.

Signed-off-by: Paul Spooren <mail@aparcar.org>

### CODE REVIEW

**Review by ukie96 — APPROVED** [review]

**Comment by aparcar** [comment]
@ukie96 did you actually test this PR? I've tried to compile it but the app doesn't seem to appear in the LuCI menu.

**Comment by aparcar** [comment]
@jow- please help me out here, I'm suspecting the `acl` parameter isn't supported in 19.07 and therefore it never shows up in the menu.

**Comment by bam80** [comment]
ping?

File: applications/luci-app-attendedsysupgrade/root/usr/share/luci/menu.d/luci-app-attendedsysupgrade.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+	"admin/system/attendedsysupgrade": {
+		"title": "Attended Sysupgrade",
+		"order": 60,
+		"action": {
+			"type": "firstchild"
+		},
+		"depends": {
+			"acl": [ "luci-app-attendedsysupgrade" ],
```
Comment (jow-) [inline applications/luci-app-attendedsysupgrade/root/usr/share/luci/menu.d/luci-app-attendedsysupgrade.json:9]: Should be `"acl": [ "attendedsysupgrade" ]`

File: applications/luci-app-attendedsysupgrade/htdocs/luci-static/resources/view/attendedsysupgrade/configuration.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+		s = m.section(form.TypedSection, 'client', _('Client'));
+		s.anonymous = true;
+
+		o = s.option(form.Flag, 'auto_search', _('Search on opening'),
+			_('Search for new sysupgrades on opening the tab'));
+		o.default = '1';
+		o.rmempty = false;
+
+		o = s.option(form.Flag, 'advanced_mode', _('Advances Mode'),
```
Comment (jow-) [inline applications/luci-app-attendedsysupgrade/htdocs/luci-static/resources/view/attendedsysupgrade/configuration.js:27]: Typo

**Comment by jow-** [comment]
You should also change the ACL definition into:

```json
{
{
	"attendedsysupgrade": {
		"description": "attended sysupgrade via rpcd and luci",
		"read": {
			"ubus": {
				"rpc-sys": [
					"upgrade_start",
					"packagelist"
				],
				"system": [
					"board",
					"info"
				],
				"uci": [
					"get"
				]
			}
		},
		"write": {
			"cgi-io": [
				"upload"
			],
			"ubus": {
				"rpc-sys": [
					"upgrade_start",
					"packagelist"
				],
				"system": [
					"board",
					"info"
				],
				"uci": [
					"set", "commit"
				]
			},
			"uci": [
				"attendedsysupgrade"
			]
		}
	}
}
```

**Comment by aparcar** [comment]
> ```json
> {
> 	"attendedsysupgrade": {
> 		"description": "attended sysupgrade via rpcd and luci",
> 		"read": {
> 			"ubus": {
> 				"rpc-sys": [
> 					"upgrade_start",
> 					"packagelist"
> 				],
> 				"system": [
> 					"board",
> 					"info"
> 				],
> 				"uci": [
> 					"get"
> 				]
> 			},
> 			"uci": [
> 				"attendedsysupgrade"
> 			]
> 		},
> 		"write": {
> 			"cgi-io": [
> 				"upload"
> 			],
> 			"ubus": {
> 				"uci": [
> 					"set", "commit"
> 				]
> 			},
> 			"uci": [
> 				"attendedsysupgrade"
> 			]
> 		}
> 	}
> }
> ```

I assume this counts not only for the backport?

**Comment by jow-** [comment]
> I assume this counts not only for the backport?

Correct.

File: applications/luci-app-attendedsysupgrade/root/usr/share/luci/menu.d/luci-app-attendedsysupgrade.json (hunk 1 above)
Comment (aparcar) [reply applications/luci-app-attendedsysupgrade/root/usr/share/luci/menu.d/luci-app-attendedsysupgrade.json:9]: Instead I renamed the acl.d file to luci-app-attendedsysupgrade to follow the style from e.g. luci-app-firewall.

File: applications/luci-app-attendedsysupgrade/htdocs/luci-static/resources/view/attendedsysupgrade/configuration.js (hunk 2 above)
Comment (aparcar) [reply applications/luci-app-attendedsysupgrade/htdocs/luci-static/resources/view/attendedsysupgrade/configuration.js]: Thanks, fixed.

**Comment by aparcar** [comment]
Tested and works on a TP-Link AC1750

**Pull request merged**